src/plugins/ethereum/ethAddress.js

```diff
@@ -0,0 +1,39 @@
+// @flow
+
+import {isAddress, toChecksumAddress} from "web3-utils";
+
+import * as C from "../../util/combo";
+import {compatibleParser} from "../../util/compat";
+
+export opaque type EthAddress: string = string;
+
+export function parseAddress(s: string): EthAddress {
+  if (!isAddress(s)) {
+    throw new Error(`not a valid ethereum address: ${s}`);
+  }
+  return toChecksumAddress(s);
+}
+
+// Utilized for a more readable and recognizable address. This is
+// needed to allow the address to fit within the Identity name requirements
+export function truncateEthAddress(address: EthAddress): string {
+  const prefix = address.slice(0, 6);
```

Comment:
Curious: is this the standard abbreviation form? e.g., for PGP keys, you
generally just take the last 8, 16, or 32 characters of the fingerprint
depending on how long you want the name to be.

If it’s standard/recognizable, no problem.

Reply:
Good question. This is an unofficial convention for rendering addresses when screen real estate is at a premium (see [metamask](https://metamask.io/) and [1hive/aragon](https://apiary.1hive.org/profile?dao=0x12b013c23f1918eac4b5342117f0b202e048566c) UIs for examples). This pull request set doesn't necessarily require this function if we just bump up the maximum name length by a few characters, which I'm actually in favor of if @decentralion gives their blessing because that way the theoretical namespace for ethereum addresses increases signficantly in size.

Even if we do bump up the name character limit, I think this should be included as a convenience function, since it needs to be included in this file due to the opaque nature of the EthAddress type.

Reply:
I approved the bump-the-size PR.